openest/generate/calculation.py

```diff
@@ -411,6 +426,10 @@ def __init__(self, region, subapp, handler, *handler_args, **handler_kw):
         self.handler_args = handler_args
         self.handler_kw = handler_kw
 
+        # Set up a buffer for results, if they don't come in at the same time
+        if isinstance(subapp, list):
```

Comment:
Good suggestion. I'm going to go with checking that `subapp` is iterable. The try/except approach strikes me as implying that we expect `subapp` to be iterable, even though in the most common cases it will not be.